src/services/TokenRegistryService.ts

```diff
@@ -0,0 +1,95 @@
+import { AxiosResponse } from "axios";
+import { CronJob } from "cron";
+import { githubApiClient } from "../axios";
+import { AssetType } from "../models/AssetType";
+import { CacheEntry, TokenRegistryServiceCache } from "../models/CacheType";
+import { GithubItem } from "../models/GitHubItem";
+import { SupportedNetworks, supportedNetworks } from "../models/Networktype";
+
+class TokenRegistryService {
+    /**
+     * The collect token data from github interval cron expression.
+     * Every hour at 0 min
+     */
+    private COLLECT_TOKEN_DATA_CRON = "0 * * * *";
+
+    private FILE_NAME_REGEX = /(?<project>\w+)-(?<id>\w+).json/;
+
+    private cache: TokenRegistryServiceCache;
+
+    constructor() {
+        this.cache = {
+            alphanet: {
+                nfts: new Map<string, CacheEntry>(),
+                nativeTokens: new Map<string, CacheEntry>()
+            },
+            testnet: {
+                nfts: new Map<string, CacheEntry>(),
+                nativeTokens: new Map<string, CacheEntry>()
+            },
+            shimmer: {
+                nfts: new Map<string, CacheEntry>(),
+                nativeTokens: new Map<string, CacheEntry>()
+            },
+            iota: {
+                nfts: new Map<string, CacheEntry>(),
+                nativeTokens: new Map<string, CacheEntry>()
+            }
+        }
+
+        this.populateCache();
+        this.scheduleCron();
+    }
+
+    public get tokenRegistryCache() {
+        return this.cache;
+    }
+
+    private populateCache() {
+        console.log("Populating cache...");
+        for (const network of supportedNetworks) {
+            void this.fetchAssetData(network, "native-tokens");
+            void this.fetchAssetData(network, "nfts");
+        }
+    }
+
+    private scheduleCron() {
+        new CronJob(this.COLLECT_TOKEN_DATA_CRON, () => this.populateCache()).start();
+    }
+
+    private async fetchAssetData(network: SupportedNetworks, assetType: AssetType) {
```

Comment:
This function doesn't cover the case in which an asset was removed from the whitelist since the cache only ever sets but never deletes any entry. Consider creating a new cache object and reset the AssetCache in the cache after it has been filled.